src/Elasticsearch/Serializer/ItemNormalizer.php

```diff
@@ -39,8 +40,10 @@ public function __construct(private readonly NormalizerInterface $decorated)
      */
     public function hasCacheableSupportsMethod(): bool
     {
-        if (!$this->decorated instanceof CacheableSupportsMethodInterface) {
-            throw new LogicException(sprintf('The decorated normalizer must be an instance of "%s".', CacheableSupportsMethodInterface::class));
+        trigger_deprecation('api-platform/core', '3.1', 'The "%s()" method is deprecated, use "getSupportedTypes()" instead.', __METHOD__);
```

Comment:
Yes, but it's the best thing we can do has the deprecation is actually introduced by Symfony. See https://github.com/api-platform/core/pull/5672#issuecomment-1643874870

Reply:
I improved the situation a bit by triggering the deprecation only if using Symfony 6.3+.

Reply:
Can’t it even wait for 3.2 then? IIRC the upstream BC layer should support it. Not 100% sure though

Reply:
As long as you use Symfony 6.3, you shouldn't use these methods. IMO we should warn the user that he's using deprecated code as soon as the alternative is available, and this will be the case in this patch release.

Reply:
Semver doesn't say anything about deprecations. TBH I don't think it is worth it to hide these deprecations just for the sake of purity.

Reply:
Well, it does 🤷‍♂️
![IMG_1782](https://github.com/api-platform/core/assets/7502063/b4786f2a-d67a-4273-9687-fcabbf4c95c4)
https://semver.org/#spec-item-7

Reply:
Indeed, I didn't remember that. But here we're just forwarding deprecation from Symfony. TBH I wouldn't bother so much about that. It will only be triggered for people using a version of Symfony that isn't compatible with the method anyway.